designer/server/src/main/scala/pl/touk/nussknacker/ui/process/newactivity/ActivityInfoService.scala

```diff
@@ -0,0 +1,59 @@
+package pl.touk.nussknacker.ui.process.newactivity
+
+import io.circe.generic.JsonCodec
+import pl.touk.nussknacker.engine.api.NodeId
+import pl.touk.nussknacker.engine.api.ProcessVersion
+import pl.touk.nussknacker.engine.api.definition.RawParameterEditor
+import pl.touk.nussknacker.engine.api.graph.ScenarioGraph
+import pl.touk.nussknacker.engine.canonicalgraph.CanonicalProcess
+import pl.touk.nussknacker.engine.definition.activity.ActivityInfoProvider
+import pl.touk.nussknacker.restmodel.definition.UiActivityParameterConfig
+import pl.touk.nussknacker.ui.process.newactivity.ActivityInfoService.{ActivityName, UiActivityNodeParameters}
+import pl.touk.nussknacker.ui.security.api.LoggedUser
+import pl.touk.nussknacker.ui.uiresolving.UIProcessResolver
+
+// TODO: move to ActivityService? execute node compilation only once with ScenarioTestService?
+class ActivityInfoService(activityInfoProvider: ActivityInfoProvider, processResolver: UIProcessResolver) {
```

Comment:
Comment about naming convention: I think that in this context the name "Scenario action" should be used. Deploy/cancel and others are scenario actions, not activities from what I know. I can be especially confusing now, when ScenarioActivities exist in different context.